src/battle_script_commands.c

```diff
@@ -3668,8 +3668,15 @@ void SetMoveEffect(bool32 primary, u32 certain)
                 break;
             case MOVE_EFFECT_TRIPLE_ARROWS:
                 {
-                    u8 randomLowerDefenseChance = RandomPercentage(RNG_TRIPLE_ARROWS_DEFENSE_DOWN, CalcSecondaryEffectChance(gBattlerAttacker, 50));
-                    u8 randomFlinchChance = RandomPercentage(RNG_TRIPLE_ARROWS_FLINCH, CalcSecondaryEffectChance(gBattlerAttacker, 30));
+                    u8 randomLowerDefenseChance;
+                    u8 randomFlinchChance;
+
+                    randomLowerDefenseChance = RandomPercentage(RNG_TRIPLE_ARROWS_DEFENSE_DOWN, 50 * CalcSecondaryEffectChance(gBattlerAttacker, gCurrentMove));
+
+                    if (GetBattlerAbility(gBattlerAttacker) == ABILITY_SERENE_GRACE && gSideStatuses[GetBattlerSide(gBattlerAttacker)] & SIDE_STATUS_RAINBOW)
```

Comment:
```suggestion
                    if (GetBattlerAbility(gBattlerAttacker) == ABILITY_SERENE_GRACE || (gSideStatuses[GetBattlerSide(gBattlerAttacker)] & SIDE_STATUS_RAINBOW))
```

Reply:
This was removed thanks to the suggestion.